lib/transformCron.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+const parser = require('cron-parser');
+const stringHash = require('string-hash');
+
+/**
+ * Evaluate a numeric hash to a number within the range of min and max
+ *
+ * @method evaluateHash
+ * @param  {Number} hash  Hash to evaluate
+ * @param  {Number} min   Minimum evaluated value
+ * @param  {String} max   Maximum evaluated value
+ * @return {Number}       Evaluated hash
+ */
+const evaluateHash = (hash, min, max) => (hash % ((max + 1) - min)) + min;
+
+/**
+ * Transform a cron value containing a valid 'H' symbol into a valid cron value
+ * @method transformValue
+ * @param  {String} cronValue   Value to transform
+ * @param  {Number} min         Minimum acceptable value
+ * @param  {Number} max         Maximum acceptable value
+ * @param  {Number} hashValue   Numeric hash to determine new value
+ * @return {String}             Transformed cron value
+ */
+const transformValue = (cronValue, min, max, hashValue) => {
+    const values = cronValue.split(',');
+
+    // Transform each ',' seperated value
+    // Ignore values that do not have a valid 'H' symbol
+    values.forEach((value, i) => {
+        // 'H' should evaluate to some value within the range (e.g. [0-59])
+        if (value === 'H') {
+            values[i] = evaluateHash(hashValue, min, max);
+
+            return;
+        }
+
+        // e.g. H/5 -> #/5
+        if (value.match(/H\/\d+/)) {
+            values[i] = value.replace('H', evaluateHash(hashValue, min, max));
+
+            return;
+        }
+
+        // e.g. H(0-5) -> #
+        if (value.match(/H\(\d+-\d+\)/)) {
+            const newMin = Number(value.substring(2, value.lastIndexOf('-')));
+            const newMax = Number(value.substring(value.lastIndexOf('-') + 1,
+                value.lastIndexOf(')')));
+
+            // Range is invalid, throw an error
+            if (newMin < min || newMax > max || newMin > newMax) {
+                throw new Error(`${value} has an invalid range, expected range ${min}-${max}`);
+            }
+
+            values[i] = evaluateHash(hashValue, newMin, newMax);
+        }
+    });
+
+    return values.join(',');
+};
+
+/**
+ * Transform a cron expression containing valid 'H' symbol(s) into a valid cron expression
+ * @method transformCron
+ * @param  {String} cronExp Cron expression to transform
+ * @param  {Number} jobId   Job ID
+ * @return {String}         Transformed cron expression
+ */
+const transformCron = (cronExp, jobId) => {
+    const fields = cronExp.split(/\s+/);
```

Comment:
Maybe should trim before this to get rid of trailing white space:
`cronExp.trim().split(/\s+/)`